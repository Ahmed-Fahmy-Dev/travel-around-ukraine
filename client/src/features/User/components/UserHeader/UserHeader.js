import React from 'react';
import { Typography, Box } from '@mui/material';
import { styled } from '@mui/material/styles';
import PropTypes from 'prop-types';

const Img = styled('img')({
  display: 'block',
  margin: 'auto 0',
  width: '80px',
  borderRadius: '50%',
  minWidth: '80px',
  height: '80px',
  backgroundColor: 'yellow',
});

const headerBoxSX = {
  display: 'flex',
  alignItems: 'start',
  m: '0 auto',
  width: '70%',
  maxWidth: '400px',
  minWidth: '300px',
};

const userNameSX = { px: { xs: '5%' }, lineHeight: '40px', my: '30px', fontSize: '30px' };

const UserHeader = ({ userData: { firstName, lastName } }) => (
  <Box sx={headerBoxSX}>
    <Img
    // src={userData.avatarUrl}
    />
    <Typography sx={userNameSX} variant="h2">{`${firstName} ${lastName}`}</Typography>
  </Box>
);

UserHeader.propTypes = {
  userData: PropTypes.object.isRequired,
};

export default UserHeader;
