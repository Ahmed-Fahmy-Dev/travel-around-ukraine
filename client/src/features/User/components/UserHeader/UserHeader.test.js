import { render } from '@testing-library/react';
import UserHeader from './UserHeader';

jest.mock('@mui/material/Typography', () => ({ children }) => <h2>{children}</h2>);
jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);

const userData = {
  firstName: 'Roman',
  lastName: 'Rymarchuk',
};

describe('Render UserHeader', () => {
  test('should UserHeader render', () => {
    const { asFragment } = render(<UserHeader userData={userData} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
