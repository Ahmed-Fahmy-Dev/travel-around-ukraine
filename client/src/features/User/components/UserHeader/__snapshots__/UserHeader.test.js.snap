// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UserHeader should UserHeader render 1`] = `
<DocumentFragment>
  <div>
    <img
      class="css-k9ebj6"
    />
    <h2>
      Roman Rymarchuk
    </h2>
  </div>
</DocumentFragment>
`;
